src/core/mount.c

```diff
@@ -1221,7 +1221,7 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         case MOUNT_UNMOUNTING_SIGKILL:
         case MOUNT_UNMOUNTING_SIGTERM:
 
-                if (f == MOUNT_SUCCESS) {
+                if (m->result == MOUNT_SUCCESS) {
```

Comment:
I don't think this is correct. We restart unmounting again and I think this should only depend on the result of last action.